src/main/java/rx/internal/operators/OperatorRetryWithPredicate.java

```diff
@@ -109,7 +109,7 @@ public void onNext(T v) {
                         };
                         // register this Subscription (and unsubscribe previous if exists) 
                         serialSubscription.set(subscriber);
-                        o.unsafeSubscribe(subscriber);
+                        o.subscribe(subscriber);
```

Comment:
I don't think so because it doesn't reuse the upstream subscriber and it decouples the subscriptions with the `SerialSubscription`.  This probably isn't the most efficient way to solve the problem but it is probably the simplest.